app/Services/StandService.php

```diff
@@ -337,104 +343,200 @@ function (Builder $query) use ($airfield) {
+                $subquery->where('groundspeed', '<=', self::MAX_OCCUPANCY_SPEED)
+                    ->where('altitude', '<=', self::MAX_OCCUPANCY_ALTITUDE);
+            })
+            ->select('network_aircraft.*')
+            ->get();
     }
 
-    public function setOccupiedStand(NetworkAircraft $aircraft): ?Stand
+    /**
+     * Go through every aircraft we're tracking and check whether or not they're occupying a stand.
+     */
+    public function setOccupiedStands(): void
     {
-        // If an aircraft cannot occupy a stand, we don't need to check any further. Vacate stand if there is one.
-        if (!$this->canOccupyStand($aircraft)) {
-            if ($aircraft->occupiedStand()->exists()) {
-                $this->vacateStand($aircraft);
+        $occupiedStandsToRemove = $this->getAircraftWithOccupiedStandsThatCanNoLongerOccupyThem();
+        $aircraftThatCanOccupyStands = $this->getAircraftThatCanOccupyStands();
+        $standOccupationsToUpdate = new Collection();
+
+        foreach ($aircraftThatCanOccupyStands as $aircraft) {
+            // Still occupying that same stand, nothing to do.
+            $currentlyOccupiedStand = $aircraft->occupiedStand->first();
+            if (
+                $currentlyOccupiedStand &&
+                $this->standOccupied($aircraft, $currentlyOccupiedStand)
+            ) {
+                continue;
             }
-            return null;
-        }
 
-        /*
-         * The aircraft is still occupying its stand.
-         */
-        if ($aircraft->occupiedStand->first() && $this->standOccupied($aircraft, $aircraft->occupiedStand()->first())) {
-            return $aircraft->occupiedStand->first();
+            $selectedStand = $this->getOccupiedStand($aircraft);
+            if ($this->occupiedStandShouldBeRemoved($currentlyOccupiedStand, $selectedStand)) {
+                $occupiedStandsToRemove->add($aircraft);
+            } elseif ($this->occupiedStandShouldBeUpdated($currentlyOccupiedStand, $selectedStand)) {
+                $standOccupationsToUpdate->add(
+                    tap(
+                        $aircraft,
+                        function (NetworkAircraft $aircraft) use ($selectedStand) {
+                            $aircraft->standToBeOccupied = $selectedStand;
+                        }
+                    )
+                );
+            }
         }
 
-        // If there's a stand that's viable, occupy the stand.
-        if ($selectedStand = $this->getOccupiedStand($aircraft)) {
-            $this->occupyStand($aircraft, $selectedStand);
-        }
+        $this->vacateStands($occupiedStandsToRemove);
+        $this->occupyStands($standOccupationsToUpdate);
+    }
 
-        return $selectedStand;
+    /**
+     * Vacate stands where required and trigger the vacation event.
+     */
+    private function vacateStands(Collection $aircraftToVacate): void
+    {
+        // Delete all the allocations
+        DB::table('aircraft_stand')
```

Comment:
Assuming this is to suppress model events?

Reply:
There's no model for aircraft_stand, and we'd have to run sync on each one individually otherwise so this is less painful I  hope :)